File: compiler/testData/asJava/lightClasses/typealiasInTypeArguments.java
public abstract interface Test /* Test*/ {
  @org.jetbrains.annotations.NotNull()
  public abstract A foo();//  foo()

  @org.jetbrains.annotations.NotNull()
  public abstract A fooAliased();//  fooAliased()

  @org.jetbrains.annotations.NotNull()
  public abstract B<A, B<A, java.lang.String>> bar();//  bar()

  @org.jetbrains.annotations.NotNull()
  public abstract B<A, B<A, java.lang.String>> barAliased();//  barAliased()
}

File: compiler/testData/asJava/lightClasses/lightClassByPsi/typealiasInTypeArguments.kt
// Test
interface A
interface B<T, R>

typealias OtherA = A
typealias OtherOtherA = OtherA
typealias OtherB<X, Y> = B<Y, X>

typealias TA = () -> Boolean

interface Test {
    fun foo(): A
    fun fooAliased(): OtherA

    fun bar(): B<A, B<A, String>>
    fun barAliased(): OtherB<OtherB<String, OtherA>, OtherOtherA>
}

class Foo(
    val p: TA?,
)
